### DISCUSSION TITLE
Bulk insert records witch batch size

### DISCUSSION THREAD

**Discussion by hipertracker** [piccolo-orm/piccolo#218] [Ideas]
Piccolo supports selecting records in the batch but for inserting records it has no option for setting a specific batch size.

**Comment by dantownsend** [comment]
You're right, I think the only solution for now would be to split it up manually:

```python
rows_to_insert = [Band(name='foo'), Band(name='bar') ...]

index = 0

while True:
    rows = rows_to_insert[index: index + 10]
    if not rows:
        break
    else:
        index += 10

    Band.insert(*rows).run_sync()
```

Having a proper API would be nicer.
